src/Core/Services/Implementations/EmergencyAccessService.cs

```diff
@@ -5,18 +5,19 @@
 using Bit.Core.Enums;
 using Bit.Core.Exceptions;
 using Bit.Core.Models;
+using Bit.Core.Models.Business.Tokens;
 using Bit.Core.Models.Data;
 using Bit.Core.Models.Table;
 using Bit.Core.Repositories;
 using Bit.Core.Settings;
-using Bit.Core.Utilities;
-using Microsoft.AspNetCore.DataProtection;
+using Bit.Core.Tokenizer;
 using Microsoft.AspNetCore.Identity;
 
 namespace Bit.Core.Services
 {
     public class EmergencyAccessService : IEmergencyAccessService
     {
+        private const string TokenKey = "EmergencyAccessServiceDataProtector";
```

Comment:
I would like to see this inside the `ITokenizer`. Feels like an implementation detail of the tokenizer to me.

Reply:
The tokenizer is responsible for encrypting the token, not for handling the key material to produce that encryption.

Ultimately this choice was to allow for keys to take the form of _either_ a service-specific key like this or an API key like is planned for cross-deployment compatible tokens.

That being said. I wasn't very happy with this either, but I can't see a way to stick this in the `Tokenizer` without creating a specific `Tokenizer` per `ITokenable` implemenation. What I'd like most is a static const on the `ITokenable` that defines the protector key, but I didn't find a way to make that work in a way I liked.

Reply:
Is that so bad then? Could we not store the `TokenKey` in `EmergencyAccessInviteToken` as a static value? And access it from the `Tokenizer`?